stdlib/public/runtime/Reflection.mm

```diff
@@ -655,9 +655,15 @@ void swift_ClassMirror_subscript(String *outString,
 #endif
   }
   
-  auto bytes = *reinterpret_cast<const char * const*>(value);
-  auto fieldData = reinterpret_cast<const OpaqueValue *>(bytes + fieldOffset);
-  
+  auto bytes = *reinterpret_cast<char * const*>(value);
+  auto fieldData = reinterpret_cast<OpaqueValue *>(bytes + fieldOffset);
+
+  if (fieldType.isWeak()) {
+    auto weakField = reinterpret_cast<WeakReference *>(fieldData);
+    auto strongValue = swift_weakLoadStrong(weakField);
+    fieldData = reinterpret_cast<OpaqueValue *>(strongValue);
+  }
```

Comment:
Doesn't this create a directness inconsistency?  That is, doesn't reflect() expect to get a T*, not a T-that-happens-to-be-a-pointer?